tensordict/memmap.py

```diff
@@ -177,6 +211,8 @@ def from_tensor(
         tensor: Union[torch.Tensor, MemmapTensor, np.ndarray],
         transfer_ownership=False,
         prefix=None,
+        filename: Optional[str] = None,
+        mode: str = "r+",
     ) -> MemmapTensor:
         if isinstance(tensor, MemmapTensor):
```

Comment:
what if I pass a memmap tensor, but i say that the filename is something different?
I would guess that the new memmap tensor is at the desired location.
Should we (1) raise an exception, (2) move the memmap tensor or (3) create a copy?

Reply:
At the moment it creates a copy at the new location.

We could certainly add a keyword argument to allow the file to be moved instead of copied though? Or an exception to be more consistent with `TensorDict.memmap_`?

Reply:
I'm not sure it does. I think it would if we'd be doing
```suggestion
        if isinstance(tensor, MemmapTensor) and filename is None:
            ...
```
but maybe i'm missing smth

Reply:
Hmm, I think you're right actually. Let me fix that and understand why my test didn't catch it.

Reply:
Thanks for catching! Should now be fixed and I've made sure the tests catch this case.